fcrepo-webapp/src/main/java/org/fcrepo/webapp/config/DatabaseConfig.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.webapp.config;
+
+import com.mchange.v2.c3p0.ComboPooledDataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+
+import javax.sql.DataSource;
+import java.beans.PropertyVetoException;
+import java.util.Map;
+
+@Configuration
+public class DatabaseConfig {
+
+    // FIXME The default location should be changed https://jira.lyrasis.org/browse/FCREPO-3337
+    @Value("${fcrepo.db.url:jdbc:h2:${java.io.tmpdir}/fcrepo-h2;FILE_LOCK=SOCKET}")
+    private String dbUrl;
+
+    @Value("${fcrepo.db.user:}")
+    private String dbUser;
+
+    @Value("${fcrepo.db.password:}")
+    private String dbPassword;
+
+    private static final Map<String, String> DB_DRIVER_MAP = Map.of(
+            "h2", "org.h2.Driver",
+            "postgresql", "org.postgresql.Driver",
+            "mariadb", "org.mariadb.jdbc.Driver",
+            "mysql", "com.mysql.cj.jdbc.Driver"
+    );
+
+    @Bean
+    public DataSource dataSource() throws PropertyVetoException {
+        final var dataSource = new ComboPooledDataSource();
+        dataSource.setDriverClass(identifyDbDriver());
+        dataSource.setJdbcUrl(dbUrl);
+        dataSource.setUser(dbUser);
+        dataSource.setPassword(dbPassword);
```

Comment:
Testing this now... but could you add a "log.debug()" statement here or somewhere else that indicates the driver and connection string being used?